### PR TITLE
TST: use frame_or_series for some tz_localize/tz_convert tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/methods/test_tz_convert.py [hunk 1]
```diff
@@ -1,24 +1,47 @@
...
 import pytest
 
-from pandas import DataFrame, Index, MultiIndex, date_range
+from pandas import DataFrame, Index, MultiIndex, Series, date_range
 import pandas._testing as tm
 
 
 class TestTZConvert:
-    def test_frame_tz_convert(self):
+    def test_tz_convert(self, frame_or_series):
```
Comment (jreback) [inline pandas/tests/frame/methods/test_tz_convert.py:9]: any appetite for moving these shared tests back to generic/methods? or else we just document that they are *always* in the frame/ tests?

Comment (jbrockmendel) [reply pandas/tests/frame/methods/test_tz_convert.py:9]: last time we discussed this we were leaning towards getting rid of generic/methods and moving those back to frame/methods

Comment (jreback) [reply pandas/tests/frame/methods/test_tz_convert.py:9]: yeah totally fine.  agreed generic is not a great place. just let's document that's where we have combined tests (doesn't have to be here).

Comment (jbrockmendel) [reply pandas/tests/frame/methods/test_tz_convert.py:9]: the test org document we made a couple months ago says that frame_or_series tests go in with the frame tests

**Pull request merged**